src/lkl/virtio_netdev.c

```diff
@@ -72,9 +72,11 @@ int lkl_virtio_netdev_add(struct virtio_dev* netdev)
         return -1;
 
     ret = dev_register(netdev);
-
     if (ret < 0)
-        sgxlkl_info("Failed to register netdev \n");
+    {
+        sgxlkl_info("Failed to register netdev\n");
```

Comment:
I think this should just be `sgxlkl_fail`.  What does the caller do with the `-1` return value?

Reply:
It's used in ``src/lkl/setup.c:1401``:

```
// Register network tap if given one
int net_dev_id = -1;
if (shm->virtio_net_dev_mem)
    net_dev_id = lkl_virtio_netdev_add(shm->virtio_net_dev_mem);
```

It looks like returning -1 is just interpreted as "no network tap present".

Reply:
Also, I've changed the info to fail as you suggested.

Reply:
I don't want to have to reason about the security implications of the network device being not present.  Simply failing is easy to reason about.

Reply:
Wait, I'm confused, do you mean that the changes are good as they are, or that I need a stronger way to fail?

Reply:
He was explaining why he wanted to fail instead of having the error bubble up.

Reply:
Confusing conversation history caused by a race condition: I wrote that before your second command and the push of the change that turned it to a fail.